mlflow/store/artifact/artifact_repo.py

```diff
@@ -319,6 +319,9 @@ def download_trace_data(self) -> Dict[str, Any]:
             - `MlflowTraceDataNotFound`: The trace data is not found.
             - `MlflowTraceDataCorrupted`: The trace data is corrupted.
         """
+        if not any(a.path == TRACE_DATA_FILE_NAME for a in self.list_artifacts()):
+            raise MlflowTraceDataNotFound(artifact_path=TRACE_DATA_FILE_NAME)
+
```

Comment:
`try_read_trace_data` on line 328 throws `MlflowTraceDataCorrupted` if the trace data is broken.